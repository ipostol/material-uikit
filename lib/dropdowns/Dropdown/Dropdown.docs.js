'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DropdownDocs = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkupComponents = require('react-markup-components');

var _Dropdown = require('./Dropdown');

var _Dropdown2 = _interopRequireDefault(_Dropdown);

var _Option = require('../Option');

var _Option2 = _interopRequireDefault(_Option);

var _Input = require('../../inputs/Input');

var _Input2 = _interopRequireDefault(_Input);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DropdownDocs = exports.DropdownDocs = function DropdownDocs() {
  return _react2.default.createElement(
    _reactMarkupComponents.Skin,
    null,
    _react2.default.createElement(
      _Dropdown2.default,
      {
        trigger: 'Open',
        contentWidth: '100px',
        onRight: true
      },
      _react2.default.createElement(
        _Option2.default,
        null,
        'Option1'
      ),
      _react2.default.createElement(
        _Option2.default,
        null,
        'Option2'
      ),
      _react2.default.createElement(
        _Option2.default,
        null,
        'Option3'
      )
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      _Dropdown2.default,
      {
        trigger: 'Open right',
        contentWidth: '300px',
        onRight: true
      },
      _react2.default.createElement(
        _Option2.default,
        { disabled: true },
        'Test Option1'
      ),
      _react2.default.createElement(
        _Option2.default,
        null,
        'Test Option2'
      )
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      _Dropdown2.default,
      {
        trigger: 'Open top',
        contentWidth: '300px',
        onTop: true
      },
      _react2.default.createElement(
        _Option2.default,
        { disabled: true },
        'Test Option1'
      ),
      _react2.default.createElement(
        _Option2.default,
        null,
        'Test Option2'
      )
    )
  );
};

exports.default = DropdownDocs;