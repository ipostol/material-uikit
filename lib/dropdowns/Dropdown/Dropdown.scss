@import '~bidstack-theme/theme';

.general {
  position: relative;
  display: inline-block;
  cursor: pointer;
}

.content {
  animation: zoom 0.3s cubic-bezier(0.31, 1, 0.34, 1);
  position: absolute;
  overflow: auto;
  padding: 4px 0;
  z-index: 10;
  width: 100px;
  top: 0;
  background-color: $white3;
  border-radius: 4px;
  box-shadow: 0 0 8px 0 rgba(0, 0, 0, 0.05), 0 4px 8px 0 rgba(0, 0, 0, 0.09);
}

.right {
  right: 0;
  transform-origin: right top;
}

.left {
  left: 0;
  transform-origin: left top;
}

.top {
  bottom: 0;
  top: auto;

  &.right {
    transform-origin: right bottom;
  }

  &.left {
    transform-origin: left bottom;
  }
}

@keyframes zoom {
  from {
    opacity: 0;
    transform: scale(0, 0);
  }

  to {
    opacity: 1;
    transform: scale(1, 1);
  }
}
