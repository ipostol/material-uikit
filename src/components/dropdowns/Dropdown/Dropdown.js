import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './Dropdown.scss';

class Dropdown extends PureComponent {

  state = {
    isOpened: false,
  };

  componentWillUnmount() {

    document.removeEventListener('click', this.listener);
    document.removeEventListener('dragstart', this.listener);

  }

  listener = () => this.close();

  open = () => {

    const { onOpen } = this.props;

    this.setState({ isOpened: true }, onOpen);
    document.addEventListener('click', this.listener);
    document.addEventListener('dragstart', this.listener);

  };

  close = () => {

    const { onClose } = this.props;

    this.setState({ isOpened: false }, onClose);
    document.removeEventListener('click', this.listener);
    document.removeEventListener('dragstart', this.listener);

  };

  render() {

    const {
      trigger,
      children,
      className,
      onOpen,
      onClose,
      onRight,
      onTop,
      contentWidth,
      ...rest
    } = this.props;

    return (
      <div
        className={cn(styles.general, className)}
        ref={ref => (this.general = ref)}
        {...rest}
      >
        <div onClick={this.open}>{trigger}</div>
        {
          this.state.isOpened && (
            <div
              className={cn(
                styles.content,
                onRight ? styles.right : styles.left,
                { [styles.top]: onTop },
              )}
              style={contentWidth && { width: contentWidth }}
            >
              {children}
            </div>
          )
        }
      </div>
    );

  }

}

Dropdown.propTypes = {
  className: PropTypes.string,
  trigger: PropTypes.node,
  children: PropTypes.node,
  onOpen: PropTypes.func,
  onClose: PropTypes.func,
  onRight: PropTypes.bool,
  onTop: PropTypes.bool,
  contentWidth: PropTypes.any,
};

export default Dropdown;
