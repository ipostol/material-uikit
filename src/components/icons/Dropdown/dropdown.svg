<?xml version="1.0" encoding="UTF-8"?>
<svg width="12px" height="8px" viewBox="0 0 12 8" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: sketchtool 42 (36781) - http://www.bohemiancoding.com/sketch -->
    <title>953080C4-7610-411D-A2B0-4A7F5798F0A2</title>
    <desc>Created with sketchtool.</desc>
    <defs></defs>
    <g id="Symbols" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="ic_expand_more" transform="translate(-6.000000, -8.000000)">
            <g>
                <rect id="Rectangle-8" x="0" y="0" width="24" height="24"></rect>
                <polygon id="Shape" fill="#FFFFFF" points="16.6 8.59997559 12 13.1999756 7.4 8.59997559 6 9.99997559 12 15.9999756 18 9.99997559"></polygon>
            </g>
        </g>
    </g>
</svg>