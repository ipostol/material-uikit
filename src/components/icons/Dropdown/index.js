import React from 'react';
import PropTypes from 'prop-types';

const image = require('./dropdown.svg');

const Dropdown = ({ style, ...rest }) => (
  <img style={{ width: '16px', marginRight: '1px', ...style }} src={image} alt="" {...rest} />
);

Dropdown.propTypes = {
  style: PropTypes.object,
};

export default Dropdown;
