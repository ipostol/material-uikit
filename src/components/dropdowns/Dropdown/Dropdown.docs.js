import React from 'react';
import { Skin } from 'react-markup-components';
import Dropdown from './Dropdown';
import Option from '../Option';
import Input from '../../inputs/Input';

export const DropdownDocs = () =>
  <Skin>
    <Dropdown
      trigger="Open"
      contentWidth="100px"
      onRight
    >
      <Option>Option1</Option>
      <Option>Option2</Option>
      <Option>Option3</Option>
      {/* <Option onClick={e => e.nativeEvent.stopImmediatePropagation()}>
        <Input value={window && window.location.href} readOnly />
      </Option> */}
    </Dropdown>
    <br />
    <br />
    <Dropdown
      trigger="Open right"
      contentWidth="300px"
      onRight
    >
      <Option disabled>Test Option1</Option>
      <Option>Test Option2</Option>
    </Dropdown>
    <br />
    <br />
    <Dropdown
      trigger="Open top"
      contentWidth="300px"
      onTop
    >
      <Option disabled>Test Option1</Option>
      <Option>Test Option2</Option>
    </Dropdown>
  </Skin>;

export default DropdownDocs;
