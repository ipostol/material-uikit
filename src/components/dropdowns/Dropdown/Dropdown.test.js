import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from './Dropdown';

describe('Dropdown component', () => {

  it('should render correctly', () => {

    const props = {};

    const wrapper = shallow(
      <Dropdown {...props} />,
    );

    wrapper.setState({ isOpened: true });

    expect(wrapper).toMatchSnapshot();

  });

  it('should render correctly onRight, contentWidth props', () => {

    const props = {
      onRight: true,
      contentWidth: 100,
    };

    const wrapper = shallow(
      <Dropdown {...props} />,
    );

    wrapper.setState({ isOpened: true });

    expect(wrapper).toMatchSnapshot();

  });

});
