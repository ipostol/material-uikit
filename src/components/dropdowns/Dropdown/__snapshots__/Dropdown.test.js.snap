// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should render correctly 1`] = `
<div
  className="general"
>
  <div
    onClick={[Function]}
  />
  <div
    className="content left"
  />
</div>
`;

exports[`Dropdown component should render correctly onRight, contentWidth props 1`] = `
<div
  className="general"
>
  <div
    onClick={[Function]}
  />
  <div
    className="content right"
    style={
      Object {
        "width": 100,
      }
    }
  />
</div>
`;
